addons/website/static/src/js/content/menu.js

```diff
@@ -233,4 +238,108 @@ publicWidget.registry.menuDirection = publicWidget.Widget.extend({
         $menu.addClass('dropdown-menu-' + alignment);
     },
 });
+
+publicWidget.registry.fadeOutHeader = publicWidget.Widget.extend({
+    selector: 'header.o_header_fade_out',
+
+    /**
+     * @override
+     */
+    start: function () {
+        this.scrollingDownwards = true;
+        this.hiddenHeader = false;
+        this.position = 0;
+        this.checkPoint = 0;
+        // Window Handlers
+        $(window).on('scroll', _.throttle(this._onWindowScroll.bind(this), 100));
```

Comment:
You have to implement a destroy method which removes that event handler. For that, add a namespace to the event:
```javascript
$(window).on('scroll.fadeOutHeader', ...)
```
then in destroy:
```javascript
$(window).off('.fadeOutHeader');
```

EDIT: in fact use the animation effect API I implemented some years ago. We only use it for the parallax animation for now. The API is not perfect but it would simplify the code here. You just have to declare the effect in the effects key and it will automatically take care of throttling and destroying the related event handlers for you. You will also automatically receive the page scroll as the first parameter of your `_onWindowScroll` function I think.